db/blob/blob_source.cc

```diff
@@ -152,6 +153,140 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
   return s;
 }
 
+void BlobSource::MultiGetBlob(
+    const ReadOptions& read_options,
+    const autovector<std::reference_wrapper<const Slice>>& user_keys,
+    uint64_t file_number, uint64_t file_size,
+    const autovector<uint64_t>& offsets,
+    const autovector<uint64_t>& value_sizes, autovector<Status*>& statuses,
+    autovector<PinnableSlice*>& blobs, uint64_t* bytes_read) {
+  size_t num_blobs = user_keys.size();
+  assert(num_blobs > 0);
+  assert(num_blobs <= MultiGetContext::MAX_BATCH_SIZE);
+  assert(num_blobs == offsets.size());
+  assert(num_blobs == value_sizes.size());
+  assert(num_blobs == statuses.size());
+  assert(num_blobs == blobs.size());
```

Comment:
We could consider passing a single `autovector` of `struct`s instead of a separate one for each of these

Reply:
Right, `std::array` requires a size that's available at compile time. You could use an `std::vector` or, even better, an `autovector` though.

Reply:
`autovector<ReadRequest> requests` looks good. We need to copy one more time when calling file_reader's `MultiGetBlob` (I believe we also need to change its `MultiGetBlob`).  But, the cost of the copy is accepted.

Reply:
This change will also need to modify `Version`, we can do it in the same new PR.  Added: https://github.com/facebook/rocksdb/issues/10156#issuecomment-1163469932

Reply:
Feel free to rename the types if the proposal is better